com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/AbortFailedProcessCommandFactory.java

```diff
@@ -66,14 +67,26 @@ private String findProcessInstanceId(CommandContext commandContext) {
         }
 
         private void abortProcessIfRequested(CommandContext commandContext, String processInstanceId) {
-            String correlationId = HistoryUtil.getVariableValue(commandContext, processInstanceId, Constants.VAR_CORRELATION_ID);
-            if (!processInstanceId.equals(correlationId)) {
-                return;
-            }
-            Boolean shouldAbortOnError = HistoryUtil.getVariableValue(commandContext, processInstanceId, Constants.PARAM_ABORT_ON_ERROR);
-            if (BooleanUtils.toBoolean(shouldAbortOnError)) {
+            if (shouldAbort(commandContext, processInstanceId)) {
                 abortProcess(commandContext, processInstanceId);
             }
+            String rootProcessInstanceId = getRootProcessInstanceId(commandContext, processInstanceId);
+            if (rootProcessInstanceId != null && shouldAbort(commandContext, rootProcessInstanceId)) {
+                abortProcess(commandContext, rootProcessInstanceId);
```

Comment:
I would suggest the following workflow -
1. Check in the current process that there is a variable PARAM_ABORT_ON_ERROR - this is done.
2. If there is -> get the correlation id
3. abort the process with the correlation id.

Benefits:
- delete the whole process instance(by the correlationId) recursively(do not rely on that logic - it could be broken). 
- No OptimisticLockingException could rise because if we are pass here we know that there is no running executions

Drawbacks:
- if there are parallel processing and we should abort - there could be an OptimisticLockingException. Check what is done in the AbortProcessAction and decide what could be done here.

Reply:
Current implementation drawbacks:
- OptimisticLocking would be risen if you have parallel exectuins because of this: `if (rootProcessInstanceId != null && shouldAbort(commandContext, rootProcessInstanceId)) {
                 abortProcess(commandContext, rootProcessInstanceId);`
- Some lefover processes will be presented because the you are not making sure that all the child executions will be deleted.
- Other elements, part of the process(jobs, executions, timers, variables, etc.), are not again guaranteed to be removed as a whole because the deletion will not happen recursively from the root process element.